dali/test/datatype_convertions.h

```diff
@@ -0,0 +1,37 @@
+#ifndef DALI_DATATYPE_CONVERTIONS_H
+#define DALI_DATATYPE_CONVERTIONS_H
+
+#include "dali/pipeline/data/tensor_list.h"
+
+namespace dali {
+
+namespace detail {
+
+template<typename Backend, typename Input>
+TensorList<Backend> ToTensorList(const Input &input, std::vector<int64_t> shape) {
+  DALI_FAIL("nope");
+};
+
+template<>
+inline TensorList <CPUBackend>
+ToTensorList<CPUBackend, std::array<float, 4ul>>(const std::array<float, 4ul> &input, std::vector<int64_t> shape) {
```

Comment:
I'm not sure to get why you constraint `input.size` to be 4 here

Reply:
This is WIP, so it's temporary just to know overall idea is ok. In final version it would be ofc generalized. It's just I'm not a pro in templates, so I need to figure out syntax here